server/runtime.go

```diff
@@ -243,7 +243,7 @@ func (e RuntimeExecutionMode) String() string {
 }
 
 type RuntimeMatchCore interface {
-	MatchInit(presenceList *MatchPresenceList, deferMessageFn RuntimeMatchDeferMessageFunction, params map[string]interface{}) (interface{}, int, error)
+	MatchInit(module string, presenceList *MatchPresenceList, deferMessageFn RuntimeMatchDeferMessageFunction, params map[string]interface{}) (interface{}, int, error)
```

Comment:
The module name should be known at match core creation time, before `MatchInit`. I suggest we pass it through the `NewRuntimeGoMatchCore` and `NewRuntimeLuaMatchCore` functions.

Additionally the match core might have a `GetHandlerName() string` function to return it, saves having to thread it through the match registry's `NewMatch` I think?